biosys/apps/main/utils_data_package.py

```diff
@@ -596,145 +592,33 @@ def find_observation_date_field_or_throw(schema):
-              "It must have a field named {} or tagged with biosys type {}. " \
-              "Alternatively you can add a foreign key the site code to extract the geometry from it. " \
-              "To create site code foreign key, add the following to you schema:{}". \
-            format(ObservationSchema.LATITUDE_FIELD_NAME,
-                   BiosysSchema.LATITUDE_TYPE_NAME,
-                   ObservationSchema.SITE_CODE_FOREIGN_KEY_EXAMPLE)
-        raise ObservationSchemaError(msg)
+    @property
+    def latitude_field(self):
+        return self.geometry_parser.latitude_field
 
-    @staticmethod
-    def find_longitude_field_or_throw(schema, enforce_required=True):
-        """
-        Precedence Rules:
-        1- Look for biosys.type = 'longitude'
-        2- Look for a field with name 'Longitude' case insensitive
-        :param schema: a dict descriptor or a Schema instance
-        :param enforce_required: if True the field must have a required constraints
-        :return: The SchemaField or raise an exception if none or more than one
-        """
-        if not isinstance(schema, GenericSchema):
-            schema = GenericSchema(schema)
-        fields = [f for f in schema.fields if f.biosys.is_longitude()]
-        if len(fields) > 1:
-            msg = "More than one Biosys longitude field found!. {}".format(fields)
-            raise ObservationSchemaError(msg)
-        if len(fields) == 1:
-            field = fields[0]
-            if enforce_required and not field.required:
-                msg = "The Biosys longitude field must be set as 'required'. {}".format(field)
-                raise ObservationSchemaError(msg)
-            else:
-                return field
-        # no Biosys longitude field found look for column name
-        fields = [f for f in schema.fields if f.name.lower() == ObservationSchema.LONGITUDE_FIELD_NAME.lower()]
-        if len(fields) > 1:
-            msg = "More than one Longitude field found!. {}".format(fields)
-            raise ObservationSchemaError(msg)
-        if len(fields) == 1:
-            field = fields[0]
-            if enforce_required and not field.required:
-                msg = "The Longitude field must be set as 'required'. {}".format(field)
-                raise ObservationSchemaError(msg)
-            else:
-                return field
-        msg = "The schema doesn't include a required longitude field. " \
-              "It must have a field named {} or tagged with biosys type {}" \
-              "Alternatively you can add a foreign key the site code to extract the geometry from it. " \
-              "To create site code foreign key, add the following to you schema:{}". \
-            format(ObservationSchema.LONGITUDE_FIELD_NAME,
-                   BiosysSchema.LONGITUDE_TYPE_NAME,
-                   ObservationSchema.SITE_CODE_FOREIGN_KEY_EXAMPLE)
-        raise ObservationSchemaError(msg)
```

Comment:
all the logic to extract geometry has been put in the new `GeometryParser` class below